app/models/framework_response.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class FrameworkResponse < VersionedModel
+  validates :type, presence: true
+
+  belongs_to :framework_question
+  belongs_to :person_escort_record
+  has_many :dependents, class_name: 'FrameworkResponse',
+                        foreign_key: 'parent_id'
+
+  belongs_to :parent, class_name: 'FrameworkResponse', optional: true
+  validates_each :value, on: :update do |record, _attr, value|
+    if value.blank? && record.parent&.option_selected?(record.framework_question.dependent_value) && record.framework_question.required
```

Comment:
This expression is a little too complicated to readily understand. Perhaps this could be split slightly, something like this....

```suggestion
      record.errors.add(:value, :blank) if value.blank? && record.requires_value?

      ...
      
  def requires_value?
      parent&.option_selected?(framework_question.dependent_value) && framework_question.required
  end
```

Reply:
I'm unsure what the behaviour of this would be if `parent` is `nil` too. Perhaps worth making that more explicit - I think this is how it will work:

```ruby
def requires_value?
  return false if parent.nil?

  parent.option_selected?(framework_question.dependent_value) && framework_question.required
end
```

Reply:
I've tidied this up and moved it into a tested method